sui-execution/latest/sui-move-natives/src/object_runtime/mod.rs

```diff
@@ -557,6 +557,20 @@ impl ObjectRuntimeState {
             })
             .collect::<Vec<_>>();
         for id in remaining_by_value_objects {
+            if let Some(owner) = input_objects.get(&id) {
+                if owner.is_shared() {
+                    return Err(ExecutionError::new(
+                        ExecutionErrorKind::SharedObjectOperationNotAllowed,
+                        Some(
+                            format!(
```

Comment:
nit, if you do `let msg = format!(...)` on another line you will probably save a few lines and some horizontal indentation. Though, definitely up to personal preference, so take it as you will